geode-cq/src/distributedTest/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceConstraintsDistributedTest.java

```diff
@@ -378,4 +418,28 @@ public boolean authorize(Method method, Object target) {
       return authorizedMethods.contains(method.getName());
     }
   }
+
+  private static class CountingCqListener implements CqListener {
+
+    private final AtomicInteger eventCount = new AtomicInteger();
```

Comment:
The fix: Use new inner class with AtomicIntegers instead of TestCqListener which is not thread safe.

Reply:
It's possible that changing to Geode and JMX APIs may have contributed to fixing the flakiness.